### PR TITLE
fix: change default tx deadline to 10m

### PR DESCRIPTION
**Pull request by just-toby**
## Description

updates the default deadline from 30 minutes (1800 seconds) to 10 minutes (600 seconds). this includes changing the constant value used to seed the default state, but also includes a migration for all users who had the old default value persisted. 

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2938/30m-feels-high-for-classic-swaps-xinwan-and-austinadams-can-weigh-in
_Relevant docs:_ https://www.notion.so/Redux-migrations-in-interface-30aca899626a4ac4af7b070f92ce976d

### QA (ie manual testing)

- [x] manually loaded the app and verified that `user.userDeadline` had the new default value saved in my IndexedDB

### Automated testing

- [x] Unit test

### CODE REVIEW

File: src/state/migrations/1.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+export type PersistAppStateV1 = {
+  _persist: PersistState
+} & { user?: UserState }
+
+/**
+ * Migration to change the default user deadline from 30 minutes to 10 minutes.
+ * We only migrate if the saved deadline is the old default.
+ */
+export const migration1 = (state: PersistAppStateV1 | undefined) => {
+  if (state?.user && state.user?.userDeadline === 1800) {
```
Comment (cbachmeier) [inline src/state/migrations/1.ts:14]: qq: we don't track if a user has set their deadline to be 1800 manually right?

File: src/state/migrations/1.test.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    userSlippageTolerance: SlippageTolerance.Auto,
+    userSlippageToleranceHasBeenMigratedToAuto: true,
+    userDeadline: 1800,
+    tokens: {},
+    pairs: {},
+    timestamp: Date.now(),
+    hideBaseWalletBanner: false,
+  },
+  _persist: {
+    version: 0,
```
Comment (cbachmeier) [inline src/state/migrations/1.test.ts:21]: nit: would this version be updated after running `migration1`? Can you check if it's updated.

File: src/state/migrations/1.test.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    rehydrated: true,
+  },
+}
+
+describe('migration to v1', () => {
+  it('should migrate the default deadline', () => {
+    const result = migration1(previousState)
+    expect(result?.user?.userDeadline).toEqual(600)
```
Comment (cbachmeier) [inline src/state/migrations/1.test.ts:29]: nit: can you also verify no other states have been changed?

**Review by cbachmeier — APPROVED** [review]
LGTM minor nits and qs

File: src/state/migrations/1.ts (hunk 1 above)
Comment (just-toby) [reply src/state/migrations/1.ts:14]: no, we just persist the deadline all the time and 1800 is the initial/default value.

File: src/state/migrations/1.test.ts (hunk 3 above)
Comment (just-toby) [reply src/state/migrations/1.test.ts:35]: yes

**Review by mattbspector — APPROVED** [review]
LGTM

**Pull request merged**